### DISCUSSION TITLE
wp_trim_words and continue reading

### DISCUSSION THREAD

**Discussion by milesdame** [wprig/wprig#728] [Q&A] [answered]
Hello, 

I'm new to WordPress development, but so far I've been okay on my own. I'm stuck though. 

I need to have a "Continue reading" button on the front page and it needs to cut off the entry content after a certain number of words. From what I've learned it appears that the more tag is not a possibility given my needs. I tried changing entry_content.php to use wp_trim_words() and get_the_content() and I thought I was heading in the right direction until I realized it was trimming the words both in the singular post and on the home page. 

Does anyone have any recommendations for where to write the code for this?

**Comment by jacklowrie** [comment, marked as answer]
If I'm understanding correctly, you want to trim the content everywhere except on the individual post page. Try wrapping the `wp_trim_words`code in a ` ! is_singular('post')` conditional.  [`is_singular()`](https://developer.wordpress.org/reference/functions/is_singular/) checks to see if the current query is for a single post of a given post type, so this will make sure that trim only fires if a user is not visiting the individual post. Let us know if that works!

**Reply by milesdame** [reply]
Thank you so much that's exactly what I needed :)
